src/Component/Tags.elm

```diff
@@ -0,0 +1,322 @@
+        AddNewTagButtonClicked ->
+            ( { model
+                | newTags =
+                    model.newTags
+                        ++ [ { name = ""
+                             , value = ""
+                             , isNew = True
+                             , isNameValidated = False
+                             }
+                           ]
+              }
+            , Task.attempt (\_ -> NoOp) (Dom.focus <| String.fromInt (List.length model.newTags) ++ "-name")
+            )
+
+        RemovedTag id_ ->
+            ( { model | newTags = List.removeAt id_ model.newTags }, Cmd.none )
+
+        TagNameChanged id_ name ->
+            let
+                ( newDebouncer, cmd ) =
+                    Debounce.push
+                        (debounceConfig DebounceTagName)
+                        ( id_, name )
+                        model.tagNameDebouncer
+            in
+            ( { model
+                | newTags = List.updateAt id_ (updateTagName name) model.newTags
+                , tagNameDebouncer = newDebouncer
+              }
+            , cmd
+            )
+
+        DebounceTagName msg_ ->
+            let
+                ( newDebouncer, cmd ) =
+                    Debounce.update
+                        (debounceConfig DebounceTagName)
+                        (Debounce.takeLast save)
+                        msg_
+                        model.tagNameDebouncer
+            in
+            ( { model | tagNameDebouncer = newDebouncer }
+            , cmd
+            )
+
+        DebouncerSettled ( id_, value ) ->
+            let
+                updateTag =
+                    updateTagName value
+                        >> (\tag ->
+                                if Maybe.isJust <| validationError model.newTags tag then
+                                    updateIsNameValidated True tag
+
+                                else
+                                    updateIsNameValidated False tag
+                           )
+            in
+            ( { model | newTags = List.updateAt id_ updateTag model.newTags }, Cmd.none )
+
+        TagValueChanged id_ value ->
+            ( { model | newTags = List.updateAt id_ (updateTagValue value) model.newTags }, Cmd.none )
+
+        TagNameOnBlur id_ ->
+            ( { model | newTags = List.updateAt id_ (updateIsNameValidated True) model.newTags }, Cmd.none )
+
+        _ ->
+            ( model, Cmd.none )
+
+
+
+-- Validation --
+
+
+validationError : List TagField -> TagField -> Maybe String
+validationError tags tag =
+    if not tag.isNew then
+        Nothing
+
+    else if String.isEmpty tag.name then
+        Just "Name of the tag can not be empty"
+
+    else if String.length tag.name > 32 then
+        Just "Name of the tag can not be longer than 32 characters"
+
+    else if List.count (\tag_ -> tag_.name == tag.name) tags > 1 then
+        Just "This name is already in use by another tag"
+
+    else
+        Nothing
```

Comment:
Right now " " is a valid tag name, and "status" and "status   " are two different tag names, but we probably don't want either.
How could we solve this?

Reply:
In which function would you put String.trim?